teps/0060-remote-resource-resolution.md

```diff
@@ -239,13 +242,113 @@ users can leverage those shared pipelines.
 
 ## Proposal
 
-<!--
-This is where we get down to the specifics of what the proposal actually is.
-This should have enough detail that reviewers can understand exactly what
-you're proposing, but should not include things like API designs or
-implementation.  The "Design Details" section below is for the real
-nitty-gritty.
--->
+Prior to moving this TEP to `implementable` there will be a considerable amount
+of experimentation required to guage suitability of different approaches. There
+are a lot of open questions wrt syntax, approach and architecture that cannot
+be easily answered without trying ideas with running code.
+
+As a first step, in
+[Pipelines#4108](https://github.com/tektoncd/pipeline/pull/4108) the controller
+is being refactored so that ref resolution occurs in reconcilers independent of
+the ones that actuate. These "resolver reconcilers" implement the exact same
+resolution behaviour as Pipelines uses today but decoupled from the core
+business logic of converting PipelineRuns to TaskRuns and TaskRuns to Pods.
+The benefits of this approach are:
+
+1. Discovery. It gives us a chance to re-review the code that makes up our
+resolution logic and root out edge cases.
+
+2. Test coverage. By isolating resolution it's easier to see where test coverage
+is incomplete.
+
+3. Libraries. Extracting resolution into packages gives us an opportunity to
+figure out which pieces can be made generally useful and which should be
+Pipelines-specific.
+
+Beyond refactoring the existing implementation there are several additional changes
+that would aid experimentation in the pre-`implementable` phase of this TEP:
+
+### Proposal 1: Create an Experimental Resolution Project
+
+This proposal proposes adding a new controller in the [Experimental
+Repo](https://github.com/tektoncd/experimental/) that implements some of
+the ideas described in the [Alternatives section](#alternatives) of this TEP.
+
+This will need a new directory added with a controller & OWNERS configuration.
+If we want to try multiple alternatives (e.g. explore both interceptor-style
+and reconciler-style approaches) then we can either have separate subdirectories
+or branches for each implementation.
+
+### Proposal 2: Disable Tekton Pipelines' "In-Tree" Resolution Logic With a Deployment Arg
```

Comment:
I think we can actually deliver this and Proposal 3 without the refactoring I mention earlier in this doc, which might end up being a __lot__ safer for existing users. I'm starting to believe that there isn't actually a way to pull the resolution logic out of the reconcilers without causing backwards-incompatible changes to the way that events are emitted, metadata is set, etc etc.

I'm going to remove the mentions of refactoring being a prerequisite here.

Reply:
specifically about emitting events, i wonder if @afrittoli 's work to move this logic to an entirely separate controller might simplify things (https://github.com/tektoncd/community/issues/435#issuecomment-854155170)